tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/upgrade-e2e-test.go

```diff
@@ -100,6 +89,17 @@ var _ = Describe(
 
 			})
 
+		})
+
+		if upgradeSuccessful {
+			cnfibuvalidations.PostUpgradeValidations()
+		}
+
+		It("Rollback successful upgrade", reportxml.ID("69058"), func() {
+			if !upgradeSuccessful {
+				Skip("Skipping rollback test due to upgrade failure.")
+			}
+
```

Comment:
cleanup should not be moved to the end of any test function as this just means the cleanup won't happen when test fails. 

Cleanup step that can potentially affect other test cases, must be added to AfterEach of the test case that creates the resource. 

Sometimes cleanup can be added to the beginning of a test function if the affecting CR may come from external sources that are out of our control.